Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -2102,39 +2111,41 @@ static void activateSessionCleanup(NetworkSessionCocoa& session, const NetworkSe
         [contexts.get() addObject:sessionWrapper.session.get()._networkContext];
     });
 
-    if (auto* clearProxies = nw_context_clear_proxiesPtr()) {
-        for (nw_context_t context in contexts.get())
-            clearProxies(context);
-    }
+    for (nw_context_t context in contexts.get())
+        clearProxies(context);
 }
 
-void NetworkSessionCocoa::setProxyConfigData(const IPC::DataReference& proxyConfigData, const IPC::DataReference& proxyIdentifierData)
+void NetworkSessionCocoa::setProxyConfigData(Vector<std::pair<Vector<uint8_t>, UUID>>&& proxyConfigurations)
 {
-    uuid_t identifier;
-    if (proxyIdentifierData.size_bytes() == sizeof(uuid_t))
-        memcpy(identifier, proxyIdentifierData.data(), proxyIdentifierData.size_bytes());
-    
-#if __has_include(<Network/proxy_config_private.h>)
-    if (auto* createProxyConfig = nw_proxy_config_create_with_agent_dataPtr())
-        m_nwProxyConfig = adoptNS(createProxyConfig(proxyConfigData.data(), proxyConfigData.size_bytes(), identifier));
-#else
-    UNUSED_PARAM(proxyConfigData);
-#endif
-    
+    auto* clearProxies = nw_context_clear_proxiesPtr();
+    auto* addProxy = nw_context_add_proxyPtr();
+    auto* createProxyConfig = nw_proxy_config_create_with_agent_dataPtr();
+    if (!clearProxies || !addProxy || !createProxyConfig)
+        return;
+
     RetainPtr<NSMutableSet> contexts = adoptNS([[NSMutableSet alloc] init]);
     forEachSessionWrapper([&contexts] (SessionWrapper& sessionWrapper) {
         if (!sessionWrapper.session)
             return;
         [contexts.get() addObject:sessionWrapper.session.get()._networkContext];
     });
+
+    for (nw_context_t context in contexts.get())
+        clearProxies(context);
+    m_nwProxyConfigs.clear();
+
+    for (auto& config : proxyConfigurations) {
+        uuid_t identifier;
+        config.second.copyToUUID_T(identifier);
```

Comment:
copyToUUID_T() treats uuid_t as a pointer type. Why is it OK to declare that pointer type on the stack here? Isn't this a pointer to nowhere?